### PR TITLE
Add monitoring labels

### PR DESCRIPTION
**Pull request by mig4**
These labels can be applied to `Service` resources to configure how
Giant Swarm Prometheus instances scrape those services.

Towards https://github.com/giantswarm/giantswarm/issues/11720

Towards https://github.com/giantswarm/giantswarm/issues/11724

## Checklist

- [ ] Consider SIG UX feedback.
- [x] Update changelog in CHANGELOG.md.
- [ ] If adding a new type, ensure that it is added to the [CRD unit tests](https://github.com/giantswarm/apiextensions/blob/d78aba91d578d56ef49f58e336035d35edc4e1b0/pkg/crd/crd_test.go#L9).

### CODE REVIEW

**Comment by mig4** [comment]
The links are 404 because the PR adding the doc is still in progress - https://github.com/giantswarm/giantswarm/pull/12512

**Comment by rossf7** [comment]
We're going to merge https://github.com/giantswarm/apiextensions/pull/520 later today with K8s 1.18 support. This will be a major version bump to v2.0.0.

@mig4 Do you want to release this before we do that?

File: pkg/label/monitoring.go [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+const MonitoringPort = "giantswarm.io/monitoring-port"
+
+// MonitoringPath is an annotation that tells Prometheus the URL path a service
+// exposes metrics on. It can be used on `Service` resources and will be picked
+// up by `simple-service` `ServiceMonitor` to configure Prometheus.
+//
+// The default is to use the path `/metrics`.
+//
+// Spec: https://intranet.giantswarm.io/docs/architecture-specs-adrs/specs/configuration-of-targets-in-tc-prometheus/
+const MonitoringPath = "giantswarm.io/monitoring-path"
```
Comment (kopiczko) [inline pkg/label/monitoring.go:27]: Maybe endpoint is a better name than a path. No big deal though.

Comment (mig4) [reply pkg/label/monitoring.go:27]: These are already in use in managed apps so while I agree the naming could be better, changing them at this point probably wouldn't be worth the effort.

Comment (kopiczko) [reply pkg/label/monitoring.go:27]: Understood :+1:

**Pull request merged**